python/nav/models/event.py

```diff
@@ -126,20 +119,17 @@ def _delete_missing_variables(self, vardict, variables):
 
     def _update_variables(self, obj, vardict):
         varmap = self._as_dict(obj)
-        related_field, var_model = self._get_model_and_related_field(obj)
 
         for key, value in vardict.items():
             if key in varmap:
                 if varmap[key].value != value:
```

Comment:
Just testing the interface manually, since our test coverage here is bad. This is now failing on line 125 (133 in the old version):

```pycon
In [36]: e.varmap = v
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-36-094e9f12244b> in <module>()
----> 1 e.varmap = v

/source/python/nav/models/event.pyc in __set__(self, obj, vardict)
     95                 self._delete_missing_variables(vardict, variables)
     96 
---> 97             self._update_variables(obj, vardict)
     98 
     99         setattr(obj, self.cachename, vardict)

/source/python/nav/models/event.pyc in _update_variables(self, obj, vardict)
    123         for key, value in vardict.items():
    124             if key in varmap:
--> 125                 if varmap[key].value != value:
    126                     varmap[key] = value
    127                     varmap[key].save()

AttributeError: 'unicode' object has no attribute 'value'
```

Reply:
I'm going to add some tests for this, just in case

Reply:
I did not change that line. I don't see how that could have worked before

Reply:
Well, naw, I take it back. As you said IRL, this has never worked - and since no production code does it, it will fly for now.